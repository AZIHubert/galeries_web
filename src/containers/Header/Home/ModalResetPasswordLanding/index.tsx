import * as React from 'react';
import {
  useDispatch,
  useSelector,
} from 'react-redux';

import Button from '#components/Button';
import Modal from '#components/Modal';
import Text from '#components/Text';

import {
  fetchSendResetPassword,
  resetSendResetPassword,
} from '#store/actions';
import {
  loadingSelector,
} from '#store/selectors';

import ModalTitle from './ModalTitle';

interface ModalResetPasswordLandingI {
  currentEmail: string;
}

const ModalResetPasswordLanding = ({
  currentEmail,
}: ModalResetPasswordLandingI) => {
  const dispatch = useDispatch();
  const loading = useSelector(loadingSelector);

  React.useEffect(() => () => {
    dispatch(resetSendResetPassword());
  }, []);

  const onClick = async () => {
    if (!loading) {
      dispatch(fetchSendResetPassword({ email: currentEmail }));
    }
  };

  return (
    <Modal.Container
      title={<ModalTitle />}
    >
      <Text>
        To reset your password, click the
        verification button in
        the email we sent to {currentEmail}.
        This helps keep your account secure.
      </Text>
      <Button.Text
        disabled={loading}
        onClick={onClick}
        styles={{
          justifyContent: 'center',
          marginBottom: 30,
          marginTop: 20,
        }}
        text='No email in your inbox or spam folder? Letâ€™s'
        textButton='resend it'
      />
    </Modal.Container>
  );
};

export default ModalResetPasswordLanding;
