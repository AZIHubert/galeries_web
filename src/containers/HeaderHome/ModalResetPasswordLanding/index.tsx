import * as React from 'react';
import {
  useDispatch,
  useSelector,
} from 'react-redux';

import Button from '#components/Button';
import Modal from '#components/Modal';
import Text from '#components/Text';

import {
  fetchSendResetPassword,
  resetSendResetPassword,
} from '#store/actions';
import {
  loadingSelector,
} from '#store/selectors';

interface ModalResetPasswordLandingI {
  currentEmail: string;
}

const ModalResetPasswordLanding = ({
  currentEmail,
}: ModalResetPasswordLandingI) => {
  const dispatch = useDispatch();
  const loading = useSelector(loadingSelector);

  React.useEffect(() => () => resetForm(), []);

  const onClick = async () => {
    if (!loading) {
      resetForm();
      dispatch(fetchSendResetPassword({ email: currentEmail }));
    }
  };

  const resetForm = () => {
    dispatch(resetSendResetPassword());
  };

  return (
    <Modal.Container
      title={(
        <Text
          textAlign='center'
        >
          Reset your password
        </Text>
      )}
    >
      <p>
        To reset your password, click the
        verification button in
        the email we sent to {currentEmail}.
        This helps keep your account secure.
      </p>
      <Button.Text
        disabled={loading}
        justifyContent='center'
        onClick={onClick}
        marginBottom={30}
        marginTop={20}
        text='No email in your inbox or spam folder? Letâ€™s'
        textButton='resend it'
      />
    </Modal.Container>
  );
};

export default ModalResetPasswordLanding;
