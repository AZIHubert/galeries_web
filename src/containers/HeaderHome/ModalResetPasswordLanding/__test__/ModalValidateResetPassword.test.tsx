import * as React from 'react';
import renderer from 'react-test-renderer';

import ThemeProvider from '#contexts/ThemeContext';

import ModalResetPasswordLanding from '../index';

describe('ModalResetPasswordLanding', () => {
  const mockedSetErrorModal = jest.fn;
  it('renders without crashing', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <ModalResetPasswordLanding
          setErrorModal={mockedSetErrorModal}
          currentEmail='mockedEmail@email.com'
        />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
