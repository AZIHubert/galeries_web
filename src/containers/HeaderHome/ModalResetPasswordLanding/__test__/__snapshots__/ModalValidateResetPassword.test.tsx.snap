// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalResetPasswordLanding renders without crashing 1`] = `
<div
  className="sc-gsTCUz gpRCqf"
  data-testid="modalValidateResetPassword"
>
  <div
    className="sc-dlfnbm erazeS"
  >
    <h3
      className="sc-hKgILt cfzZCa"
    >
      Reset your password
    </h3>
    <p
      data-testid="validateResetPasswordBody"
    >
      To reset your password, click the verification button in the email we sent to 
      mockedEmail@email.com
      . This helps keep your account secure.
    </p>
    <div
      className="sc-gKsewC hQKVAW"
      fontSize={1}
    >
      <p>
        No email in your inbox or spam folder? Letâ€™s 
        <span
          className="sc-jSgupP eLLNoK"
          onClick={[Function]}
        >
          resend it
        </span>
        .
      </p>
    </div>
  </div>
  <div
    className="sc-bdfBwQ kUovzw background-container"
  />
</div>
`;
